### DISCUSSION TITLE
Themes with structural differences

### DISCUSSION THREAD

**Discussion by kagiorgis** [tailwindlabs/tailwindcss#9575] [Help]
_This is a follow up to a [previous post](https://github.com/tailwindlabs/tailwindcss/discussions/9564)._

Except for the light and dark themes, I want to have a third option, where not only colours etc. change, but rather the entire structure of some elements such as buttons etc. so I guess creating variables for classes wouldn't make sense. 

The way I'm doing it for now is to have 3 theme options:

- For the **light** and **dark** modes I simply use the built-in dark mode, eg: `<h1 className:"test-black dark:text-white">Title</h1>`

- For my third theme option "**retro**", since many components have different structures that cannot be replicated within the elements as they are in the light/dark mode, I have decided to have 2 versions of the given page within the same .jsx file split in 2 parent divs.
```
<div className="retro-hidden">Content for Light and Dark modes</div>
<div className="hidden retro-block">Same Content but for Retro mode</div>
```

The retro-hidden & retro-block classNames refer to the following CSS rules:
`.retro {
    .retro-block {
      display: contents;
    }
    .retro-hidden {
      display: none;
    }
  }`

So far everything works fine, but makes my workflow much slower, since I have to carefully replicate everything for the retro mode, and also doing it this way feels a bit silly.
Thinking about it, there has to be a way that's faster, safer and makes more sense. 

Any ideas? I 'm still new to this, so please excuse my question if it's super-silly.

**Comment by wongjn** [comment]
You say "many components have different structures" but then you "have to carefully replicate everything for the retro mode" which slightly contradicts each other. Could it be that some smaller components have different structure but the larger-scale page structure is similar across themes? Could you not then only do the swap out at the lowest level you can? And since you are using React, you could take advantage of it to swap out the DOM as needed. For example, instead of this:
```jsx
<div className="retro-hidden">
  <h2>Foo title</h2>
</div>
<div className="hidden retro-block">
  <h1>Bar title</h1>
</div>
```
Do this:
```jsx
isRetro() ? <h1>Bar title</h1> : <h2>Foo title</h2>
```

**Reply by kagiorgis** [reply]
Yes, indeed it's smaller components that have a different structure, buttons or submenus for example.
Is the way I'm currently doing it bad practice?

**Reply by wongjn** [reply]
I am not well enough informed to determine if it is bad practice, but I can say that as you have come to find out, it has made your workflow much slower. Plus you're having redundant DOM nodes sitting around too which doesn't feel right to me.

**Reply by kagiorgis** [reply]
True, thanks for the advice, I 'll play around with your solution and see if it makes sense for my project.
Cheers :)
